### DISCUSSION TITLE
OpenWrt 路由器插件 `luci-app-CloudflareSpeedTest` 自动更新 `SSR+ / PassWall` 节点 IP

### DISCUSSION THREAD

**Discussion by mingxiaoyu** [XIU2/CloudflareSpeedTest#319] [分享]
![image](https://user-images.githubusercontent.com/19354202/153700655-6d37e6fb-78da-40eb-8dc2-636645fb1e7f.png)

项目地址： https://github.com/mingxiaoyu/luci-app-cloudflarespeedtest

**Comment by goukey** [comment]
这个怎么用？意思是给你的翻墙节点测一个最快的ip？？

**Comment by XIU2** [comment]
@goukey 类似于把 #156 这种脚本改成了 OpenWrt 插件来方便使用，不过我对 OpenWrt 不太了解~

**Comment by goukey** [comment]
我还是没看懂，如果是单纯给节点地址选出一个最快的cdn地址，那么不就意味着对国内中转的节点毫无作用吗。

@XIU2 还是说这个脚本的意思是把远程要走代理的地址也筛选出一个最快的？？比如我用ssr+访问Google.com，节点出口在香港。脚本能测出Google香港最快的ip？？

**Comment by XIU2** [comment]
@goukey 首先，CloudflareST 的作用是帮你快速找到**速度最快的 Cloudflare CDN IP**（相对于你而言）。

> 这个插件及 #156 的脚本，它们只适用于**套了 Cloudflare CDN 的代理**。

一些人为了**复活被墙 IP** 的服务器，或为了**预防被墙**，或为了**拯救线路辣鸡**的服务器，会将自己的代理服务器套上了 Cloudflare CDN。因此如果要想要提高代理速度，就需要用到 CloudflareST 测速获得最快 IP，以前是只能**手动测速**后，再去**手动修改节点 IP**（即 Cloudflare CDN 的 IP），而这些插件/脚本则会**自动测速并帮你替换节点 IP**。

``` yaml
# 直连代理：
你 <=> 代理服务器 <=> 目标网站
# 套了 Cloudflare CDN 的代理：
你 <=> Cloudflare CDN <=> 代理服务器 <=> 目标网站
```

如果你的代理没有套 Cloudflare CDN，那么这个插件 及 脚本对你来说**完全无用**。

**Reply by iamchq** [reply]
这个项目已经一年多没更新了，看到有个fork的这个的新项目，并在原版基础上进行了大量的重构与改进，项目地址：https://github.com/stevenjoezhang/luci-app-cloudflarespeedtest ，里面也宣传致谢了原作者，希望可以在衍生项目一栏添加下

**Comment by mingxiaoyu** [comment]
> 我还是没看懂，如果是单纯给节点地址选出一个最快的cdn地址，那么不就意味着对国内中转的节点毫无作用吗。
> 
> @XIU2 还是说这个脚本的意思是把远程要走代理的地址也筛选出一个最快的？？比如我用ssr+访问Google.com，节点出口在香港。脚本能测出Google香港最快的ip？？

有钱的用的国内中转。

这个是个白嫖党的和复活vps用的 CDN加速

**Comment by likefu40** [comment]
厉害啊！！

**Comment by likefu40** [comment]
> > 我还是没看懂，如果是单纯给节点地址选出一个最快的cdn地址，那么不就意味着对国内中转的节点毫无作用吗。
> > @XIU2 还是说这个脚本的意思是把远程要走代理的地址也筛选出一个最快的？？比如我用ssr+访问Google.com，节点出口在香港。脚本能测出Google香港最快的ip？？
> 
> 有钱的用的国内中转。
> 
> 这个是个白嫖党的和复活vps用的 CDN加速

这就是，代理节点套cdn的，优选IP工具。因为套了CDN，优选出本地到cdn最快的IP，那就变相加速了科学上网

**Comment by zhjan33** [comment]
使用量不多，但装上这插件和依赖第二天，CF请求量己超出免费额度。

**Comment by xingaowei** [comment]
牛逼啊，这个都写出来了
